client/lib/use-in-view/index.ts

```diff
@@ -1,21 +1,20 @@
 import { RefObject, useEffect, useRef } from 'react';
 
 export function useInView< T extends Element >( oneTimeCallback: () => void ): RefObject< T > {
-	const viewedRef = useRef< boolean >( false );
+	const viewedRef = useRef( false );
 	const elementRef = useRef< T >( null );
-	const observerRef = useRef< IntersectionObserver >();
+	const oneTimeCallbackRef = useRef< () => void >();
+
+	useEffect( () => {
+		oneTimeCallbackRef.current = oneTimeCallback;
+	}, [ oneTimeCallback ] );
```

Comment:
This is OK but useless, we can just set the reference on render without the effect since there's not really a performance penalty with this assignment. Furthermore, it might actually prevent race conditions between updated versions of the function and the actual listener inside the effect below. I'd recommend unwrapping it.

```suggestion
	oneTimeCallbackRef.current = oneTimeCallback;
```

Reply:
Fixed.